test/test-functions

```diff
@@ -626,7 +626,8 @@ inst_libs() {
 }
 
 import_testdir() {
-    STATEFILE=".testdir"
+    STATEFILE="${BUILD_DIR:-.}/test/$(basename $(dirname $(realpath $0)))/.testdir"
+    mkdir -p $(dirname $STATEFILE)
```

Comment:
`make clean` can't seem to remove `.testir`, so the other occurrences of `.testdir` should be probably replaced as well.
```sh
$ grep '\.testdir' test/test-functions
    STATEFILE="${BUILD_DIR:-.}/test/$(basename $(dirname $(realpath $0)))/.testdir"
                rm -f .testdir
                    rm -f .testdir
```